lib/matplotlib/scale.py

```diff
@@ -102,7 +102,11 @@ def __init__(self, axis):
 
     def set_default_locators_and_formatters(self, axis):
         # docstring inherited
-        axis.set_major_locator(AutoLocator())
+        if (axis.axis_name == 'x' and (mpl.rcParams['xtick.top'] or mpl.rcParams['xtick.bottom']) or
+                axis.axis_name == 'y' and (mpl.rcParams['ytick.left'] or mpl.rcParams['ytick.right'])):
+            axis.set_major_locator(AutoLocator())
+        else:
+            axis.set_major_locator(NullLocator())
```

Comment:
I think this is breaking on polar where I think these axes are named something else (r and theta?)

But I'm curious how this is helping you?  Are your axes not named x and y?

Reply:
> But I'm curious how this is helping you? Are your axes not named x and y?

This logic is copied directly from below (109-113 on `main`), only adapting `major` in place of `minor`. Below for the minor ticks if the axis is x (or y), and the user has `x.minor.visible` set to `False`, it uses `NullLocator`. So this was the equivalent logic (I think) where if the axis is x (or y), and the user has `xtick.visible` set to `False`, it uses `NullLocator`.

Reply:
> I think this is breaking on polar where I think these axes are named something else (r and theta?)

I think I see something else. Assuming `Polar` uses a `Linear` scaling (I assume it does), `Polar` could be making an assumption that the locator is `AutoLocator` or whatever the default used to be before my change. Then you `NullLocator.set_params(...)` which is not allowed (safely at least). At least that's what I'm thinking from this traceback:
```
/home/runner/work/matplotlib/matplotlib/lib/matplotlib/projections/polar.py:421: in clear
    super().clear()
/home/runner/work/matplotlib/matplotlib/lib/matplotlib/axis.py:897: in clear
    self._set_scale('linear')
/home/runner/work/matplotlib/matplotlib/lib/matplotlib/projections/polar.py:433: in _set_scale
    self.get_major_locator().set_params(steps=[1, 1.5, 3, 4.5, 9, 10])
/home/runner/work/matplotlib/matplotlib/lib/matplotlib/ticker.py:1605: in set_params
    _api.warn_external(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

message = "'set_params()' not defined for locator of type <class 'matplotlib.ticker.NullLocator'>"
```
One easy fix/workaround is to just make `clear` or maybe better `Polar._set_scale` set the formatter to `AutoFormatter` since it almost immediately thereafter assumes it can `set_params` on it.

Reply:
OK, I see - this is just not setting the locators/formatters if the ticks are turned off.  That seems a reasonable short path, if a rare use case.